jwst/associations/lib/rules_level3_base.py

```diff
@@ -478,6 +478,21 @@ def ok_candidate(self, member=None):
         """
         return self.acid.type.lower() not in INVALID_AC_TYPES
 
+    def finalize(self):
+        if self.acid.type.lower() == 'group':
+            if len(self.current_product['members']):
+                try:
+                    # Janky extraction of obs_num from filenames
```

Comment:
"Janky"  Is that a technical term?

Reply:
I like "janky".

However, there is a more "association-like" way of doing this. There is an "informational" constraint, called `Constraint_Obsnum`, that is part of the `AsnMixin_Science` class. The whole point of this constraint is to just gather all the observation numbers that are part of the association.

The multi-observation check can then be to ensure that the list of `self.constraints.['obs_num'].found_values` is greater than one.

Reply:
This will take away the filename dependency, and make it contolled by the input association pool.

Reply:
Some further thoughts:
- This method should be moved to `Asn_MixinScience`
- Ensure that parent class `finalize` is also called, adding the `super` call at the end of the method.